app/controllers/hosts_controller.rb

```diff
@@ -479,6 +479,15 @@ def update_multiple_power_state
     redirect_back_or_to hosts_path
   end
 
+  def show_multiple_power_status
+    find_multiple_ajax
+    res = {}
+    @hosts.includes(:compute_resource).each do |host|
+      res[host.id] = host.supports_power_and_running? rescue false
```

Comment:
Another suggestion would be to format the html here, before sending it to the client. Then the client would be able to use the rendered html without generating it on the client side.

I would add a method to `hosts_helper.rb` that would generate the template and then put the result as a string value into our json:

``` ruby
host_power_status = :running # should be one of three: :running, :shut_down, :unknown
res[host.id] = power_status_template(host_power_status)
```

Then in `host_power.js` just replace the spinner with the HTML you got from the JSON.